packages/flutter_tools/lib/src/build_system/source.dart

```diff
@@ -244,3 +278,17 @@ class _ArtifactSource implements Source {
   @override
   bool get implicit => false;
 }
+
+class _HostArifactSource implements Source {
+  const _HostArifactSource(this.artifact, { this.platform, this.mode });
+
+  final HostArtifact artifact;
+  final TargetPlatform platform;
+  final BuildMode mode;
```

Comment:
Why does `_HostArifactSource` have a `BuildMode`, isn't the lack what differentiates the host from non-host artifacts?

Reply:
done

Reply:
(its not used, removed)